### PR TITLE
Pumas nt2 bigg v2

### PR DESCRIPTION
**Pull request by cacraigucar**
Add refl10cm and reflz10cm diagnostics

Tested with ERP_D_Ln9_Vnuopc.ne30pg3_ne30pg3_mg17.F2000dev.cheyenne_intel.cam-outfrq9s

### CODE REVIEW

**Comment by sjsprecious** [comment]
@cacraigucar Thanks for this PR. I have a few minor comments related to the OpenACC directives. Should I commit directly to your branch or leave a comment here?

**Comment by cacraigucar** [comment]
> 

There are two ways to move forward (since @andrewgettelman tested with this).

1. We go ahead and commit my changes as is and then @sjsprecious makes his changes directly in PUMAS.  @Katetc makes a tag after the two commits are done.
2. @sjsprecious opens a PR back to my branch, I bring the changes in and we commit them all together to PUMAS

Since @andrewgettelman has tested and I'd like to make sure the answers don't change from that run since he won't be able to run another test for a number of weeks, I prefer 1.  That way I can make sure answers don't change.  Thoughts?

**Comment by Katetc** [comment]
If the notes are just related to ACC directives, then they won't change answers in a standard CPU run. I'm curious to see if his issues are related to the merge after his tag. Jian, could you please put your comments in this PR. If they are minimal, we can get them ironed out for this tag.

File: micro_pumas_v1.F90 [hunk 1]
```diff
@@ -1240,15 +1245,15 @@ subroutine micro_pumas_tend ( &
   !$acc               pgsacwtot,pgracstot,prdgtot,qmultgtot,qmultrgtot,       &
   !$acc               psacrtot,npracgtot,nscngtot,ngracstot,nmultgtot,        &
   !$acc               nmultrgtot,npsacwgtot,nrout,nsout,refl,arefl,           &
-  !$acc               areflz,frefl,csrfl,acsrfl,fcsrfl,rercld,ncai,ncal,      &
+  !$acc               areflz,frefl,csrfl,acsrfl,fcsrfl,refl10cm,reflz10cm,rercld,ncai,ncal,      &
```
Comment (sjsprecious) [inline micro_pumas_v1.F90:1248]: I would suggest aligning the openacc directives by the `&` sign. If you think it is too tedious to change every line below, maybe just move `refl10cm` and `reflz10cm` after `nmeltgtot`.

File: micro_pumas_v1.F90 [hunk 2]
```diff
@@ -4252,9 +4268,70 @@ subroutine micro_pumas_tend ( &
            acsrfl(i,k)=0._r8
            fcsrfl(i,k)=0._r8
         end if
+     end do
+  end do
+
```
Comment (sjsprecious) [inline micro_pumas_v1.F90:4273]: Add `!$acc end parallel` directive here.

File: micro_pumas_v1.F90 [hunk 3]
```diff
@@ -4181,14 +4189,23 @@ subroutine micro_pumas_tend ( &
...
 
         ! analytic radar reflectivity
         !--------------------------------------------------
         ! formulas from Matthew Shupe, NOAA/CERES
         ! *****note: radar reflectivity is local (in-precip average)
         ! units of mm^6/m^3
 
+     end do
+  end do
+
```
Comment (sjsprecious) [inline micro_pumas_v1.F90:4202]: Add `!$acc end parallel` here.

File: micro_pumas_v1.F90 [hunk 4]
```diff
@@ -4181,14 +4189,23 @@ subroutine micro_pumas_tend ( &
...
+  ! Min rain rate of 0.1 mm/hr
+  rthrsh=0.0001_r8/3600._r8
+
```
Comment (sjsprecious) [inline micro_pumas_v1.F90:4205]: Add `!$acc parallel vector_length(VLENS) default(present)` here.

File: micro_pumas_v1.F90 [hunk 5]
```diff
@@ -4252,9 +4268,70 @@ subroutine micro_pumas_tend ( &
...
+  !--------------------------------------------------
+  ! Formula from Hugh Morrison
+  ! Ice dielectric correction from Smith 1984, Equation  10 and Snow correction from Smith 1984 Equation 14
+  ! Smith, Paul L. “Equivalent Radar Reflectivity Factors for Snow and Ice Particles.
+  !                ” Journal of Climate and Applied Meteorology 23, no. 8 (1984): 1258–60.
+  !                DOI:  10.1175/1520-0450(1984)023<1258:ERRFFS>2.0.CO;2
+
+  ! *****note: radar reflectivity is local (in-precip average)
+  ! units of mm^6/m^3
+
```
Comment (sjsprecious) [inline micro_pumas_v1.F90:4284]: Add `!$acc parallel vector_length(VLENS) default(present)` here.

File: micro_pumas_v1.F90 [hunk 6]
```diff
@@ -4181,14 +4189,23 @@ subroutine micro_pumas_tend ( &
...
         ! *****note: radar reflectivity is local (in-precip average)
         ! units of mm^6/m^3
 
+     end do
+  end do
+
+  ! Min rain rate of 0.1 mm/hr
+  rthrsh=0.0001_r8/3600._r8
+
+  !$acc loop gang vector collapse(2) private(dum,dum1)
```
Comment (sjsprecious) [inline micro_pumas_v1.F90:4206]: Remove `private(dum,dum1)`.

File: micro_pumas_v1.F90 [hunk 7]
```diff
@@ -4252,9 +4268,70 @@ subroutine micro_pumas_tend ( &
...
+  ! Formula from Hugh Morrison
+  ! Ice dielectric correction from Smith 1984, Equation  10 and Snow correction from Smith 1984 Equation 14
+  ! Smith, Paul L. “Equivalent Radar Reflectivity Factors for Snow and Ice Particles.
+  !                ” Journal of Climate and Applied Meteorology 23, no. 8 (1984): 1258–60.
+  !                DOI:  10.1175/1520-0450(1984)023<1258:ERRFFS>2.0.CO;2
+
+  ! *****note: radar reflectivity is local (in-precip average)
+  ! units of mm^6/m^3
+
+  !$acc loop gang vector collapse(2) private(dum,dum1)
```
Comment (sjsprecious) [inline micro_pumas_v1.F90:4285]: Remove `private(dum,dum1)`.

File: micro_pumas_v1.F90 [hunk 8]
```diff
@@ -4252,9 +4268,70 @@ subroutine micro_pumas_tend ( &
...
+        refl10cm(i,k) = 10._r8*dlog10(dum)
+
+     end do
+  end do
+
+  !redefine fice here....
+
+  !$acc loop gang vector collapse(2)
+  do k=1,nlev
+     do i=1,mgncol
```
Comment (sjsprecious) [inline micro_pumas_v1.F90:4334]: Remove these lines and only keep the `!redefine fice here` comment.

**Review by sjsprecious — CHANGES_REQUESTED** [review]

**Comment by sjsprecious** [comment]
Hi @cacraigucar and @Katetc , thanks for your comments. I just left the comments at the places where I think some GPU directive changes are needed. They are not answer-changing either on CPU or GPU, but really a requirement for a consistent coding style. I think they can go in as the same tag as Cheryl's changes here but let me know if you have any questions.

**Review by Katetc — APPROVED** [review]

**Pull request merged**